sources/exec_text_tree.c

```diff
@@ -6,46 +6,52 @@
 /*   By: inikulin <inikulin@student.42berlin.de>    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/09/23 13:39:01 by inikulin          #+#    #+#             */
-/*   Updated: 2024/10/10 01:57:42 by inikulin         ###   ########.fr       */
+/*   Updated: 2024/10/12 02:27:29 by inikulin         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "minishell.h"
 
-static char	*execute_text_tree_node(t_param *param, t_treenode *node)
+static int	is(char *a, char *b)
 {
-	char				*out;
-	char				*cmd;
-	t_execution_result	res;
+	return (ft_strcmp(a, b));
+}
+
+static int	exec_rec(t_param *param, t_treenode *node)
+{
+	int		res;
+	t_tree	t;
 
-	cmd = node->content;
-	out = 0;
-	option_cd(ft_strncmp(cmd, "cd", 3) == 0, node, param, &out);
-	option_echo(!out && ft_strncmp(cmd, "echo", 5) == 0, node, param, &out);
-	option_env(!out && ft_strncmp(cmd, "env", 4) == 0, node, param, &out);
-	option_exit(!out && ft_strncmp(cmd, "exit", 5) == 0, node, param, &out);
-	option_export(!out && ft_strncmp(cmd, "export", 7) == 0, node, param, &out);
-	option_pwd(!out && ft_strncmp(cmd, "pwd", 4) == 0, node, param, &out);
-	option_unset(!out && ft_strncmp(cmd, "unset", 6) == 0, node, param, &out);
-	option_external(!out, node, param, &res);
-	if (!out)
-		printf("%s: %s\n", cmd, ERR_COMMAND_NOT_FOUND);
-	else
-		printf("%s\n", out);
-	return (out);
+	if (!node)
+		return (0);
+	if (!ft_strcmp(node->content, "("))
+		return (exec_rec(param, node->child));
+	t.root = node;
+	if (param->debug_output_level & DBG_PRINT_NODE_BEFORE_EXEC)
+		ft_tree_print_s(&t);
+	res = execute_text_tree_node(param, node);
+	node = node->sibling_next;
+	while (!res && node && !is(node->content, "&&") && node->sibling_next)
```

Comment:
skipping node and operators if the command's execution failed, ok, nice

Reply:
If a command returned false, this means all other commands following *that are connected with &&* will not be executed.